x-pack/plugins/infra/public/pages/logs/log_entry_categories/page_setup_content.tsx

```diff
@@ -4,99 +4,54 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { EuiSpacer, EuiSteps, EuiText } from '@elastic/eui';
+import React, { useState, useCallback } from 'react';
 import { FormattedMessage } from '@kbn/i18n/react';
-import React, { useMemo } from 'react';
+import { EuiText, EuiButton, EuiSpacer } from '@elastic/eui';
 
-import { BetaBadge } from '../../../components/beta_badge';
 import {
-  createInitialConfigurationStep,
-  createProcessStep,
   LogAnalysisSetupPage,
   LogAnalysisSetupPageContent,
   LogAnalysisSetupPageHeader,
 } from '../../../components/logging/log_analysis_setup';
 import { useTrackPageview } from '../../../../../observability/public';
-import { useLogEntryCategoriesSetup } from './use_log_entry_categories_setup';
+import { LogEntryCategoriesSetupFlyout } from './setup_flyout';
 
 export const LogEntryCategoriesSetupContent: React.FunctionComponent = () => {
   useTrackPageview({ app: 'infra_logs', path: 'log_entry_categories_setup' });
   useTrackPageview({ app: 'infra_logs', path: 'log_entry_categories_setup', delay: 15000 });
 
-  const {
-    cleanUpAndSetUp,
-    endTime,
-    isValidating,
-    lastSetupErrorMessages,
-    setEndTime,
-    setStartTime,
-    setValidatedIndices,
-    setUp,
-    setupStatus,
-    startTime,
-    validatedIndices,
-    validationErrors,
-    viewResults,
-  } = useLogEntryCategoriesSetup();
-
-  const steps = useMemo(
-    () => [
-      createInitialConfigurationStep({
-        setStartTime,
-        setEndTime,
-        startTime,
-        endTime,
-        isValidating,
-        validatedIndices,
-        setupStatus,
-        setValidatedIndices,
-        validationErrors,
-      }),
-      createProcessStep({
-        cleanUpAndSetUp,
-        errorMessages: lastSetupErrorMessages,
-        isConfigurationValid: validationErrors.length <= 0 && !isValidating,
-        setUp,
-        setupStatus,
-        viewResults,
-      }),
-    ],
-    [
-      cleanUpAndSetUp,
-      endTime,
-      isValidating,
-      lastSetupErrorMessages,
-      setEndTime,
-      setStartTime,
-      setUp,
-      setValidatedIndices,
-      setupStatus,
-      startTime,
-      validatedIndices,
-      validationErrors,
-      viewResults,
-    ]
-  );
+  const [isFlyoutOpen, setIsFlyoutOpen] = useState<boolean>(true);
+  const openFlyout = useCallback(() => setIsFlyoutOpen(true), []);
+  const closeFlyout = useCallback(() => setIsFlyoutOpen(false), []);
```

Comment:
Should we move this a level up to make sure it persists across setup and results pages? The way it is right now the user never sees the "done, click to view the results" stage, because the flyout visibility is lost as the page switches from setup content to results content.

This loss of state even happens twice when an existing job is recreated: Once after the old job is deleted and once after the new job has been created.

Could be a hook shared via the context or just props passed down from the parent.

Reply:
Wouldn't it make sense then to move also the `<*SetupFlyout />` components one level higher?

Reply:
Update: I moved it to the page content, but I'm not sure it is a good idea. There are new and interesting edge cases caused by the difference between `jobStatus` and `setupStatus`. For example if you leave the flyout open long enough after the first setup, it will ask the user if it wants to create the job again, or if you try to recreate the job from the results page, once you try to create the job it will briefly show the setup.

I wonder what's more desirable here:
1. Revert the last commit and render the flyout back to the setup and results pages directly, handing the state there.
2. Workaround the issues.
3. Go back to [this approach](https://github.com/elastic/kibana/pull/68366/commits/f3b80b337cefb1f402ac900ba24acb828322e092), where the visibility of the flyout is handled with `setupStatus`.

I think 1 is faster, and I find 3 cleaner than 2, because we handle state in only one place.

what do you think?

Reply:
The problem with option 3 is that it won't be clean anymore once we have more than one module managed by the page (which will be the case within a few days).

I'll have to ponder this for a few minutes... :thinking:

Reply:
After trying it out it feels pretty reasonable the way you wrote it right now if the following situation is taken care of:

> if you leave the flyout open long enough after the first setup, it will ask the user if it wants to create the job again

That happens because the reducer transitions the state matchine from `succeeded` to `skipped` when the job state is polled in the background. This could be avoided by making sure in `getSetupStatus` that the transition from `succeeded` to `skipped` is only performed in reaction to the `viewedResults` action (which should also be dispatched when the flyout is closed via the "x" after a successful setup).

> if you try to recreate the job from the results page, once you try to create the job it will briefly show the setup

That doesn't sound like a problem, does it?

Reply:
Alright, I think I got the flow correctly right now.